Bottom: 30dd87431c6508a0c0ad3f592ccc75881c4b0a66
Top:    88ab500f79e2b58e6f0d38c4e4432f7c7a4657d7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-11 16:01:43 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 79144f7..353c59f 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1160,7 +1160,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
